### ISSUE TITLE
unrealized.py: "A valid price could not be found" warnings for leaked cost basis

### ISSUE THREAD

**Issue by blais** [beancount/beancount#16]
**[Original report](https://bitbucket.org/blais/beancount/issue/16) by Matthew Harris (Bitbucket: [matthew_harris](https://bitbucket.org/matthew_harris), GitHub: [mharris312](https://github.com/mharris312)).**

----------------------------------------

Quoting yourself:
As for the 
"A valid price for ABC/USD could not be found"
message: it is emitted from the unrealized gains plugin.
I'll have to hunt it down, sounds like a bug.

This is indeed, a bug triggered by this situation and I'll have to fix it.
The unrealized gains plugin computes the list of holdings in order to insert the gains, and it ends up with a holding object like this for your leaked cost basis:

```
#!python

Holding(account='Assets:Investments:RolloverIRA:ABC', number=Decimal('0.000'), currency='ABC', cost_number=None, cost_currency='USD', book_value=Decimal('-8222.48552000000000000000000'), market_value=None, price_number=None, price_date=datetime.date(2013, 4, 1))

```

Now, it expects that if the commodity is not a currency, it will have a price. But price_number is None, so it complains about it.
I'll fix this over the weekend; it should be silent when that occurs, I'll have to find conditions for that.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

I will ignore those holdings, thereby ignoring the leaked cost basis. This should only be able to occur when "NONE" booking is used, so I thin, it's a fair treatment.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Actually, no, I cannot do that, because of the way merged holdings will be treated.
I'll adjust the unrealized plugin instead.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #16, the unrealized plugin should not warm on leaked cost basis.
